crates/common/fork_choice/src/store.rs

```diff
@@ -295,4 +309,280 @@ impl Store {
             Ok(head_root)
         }
     }
+
+    pub fn update_latest_messages(
+        &mut self,
+        attesting_indices: Vec<u64>,
+        attestation: Attestation,
+    ) -> anyhow::Result<()> {
+        let target = attestation.data.target;
+        let beacon_block_root = attestation.data.beacon_block_root;
+        let mut non_equivocating_attesting_indices = vec![];
+
+        for &index in &attesting_indices {
+            if !self.equivocating_indices.contains(&index) {
+                non_equivocating_attesting_indices.push(index);
+            }
+        }
+
+        for index in &non_equivocating_attesting_indices {
+            if !self.latest_messages.contains_key(index)
+                || target.epoch
+                    > self
+                        .latest_messages
+                        .get(index)
+                        .ok_or(anyhow!(
+                            "Could not get expected latest message at index: {index}"
+                        ))?
+                        .epoch
+            {
+                self.latest_messages.insert(
+                    *index,
+                    LatestMessage {
+                        epoch: target.epoch,
+                        root: beacon_block_root,
+                    },
+                );
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn on_tick_per_slot(&mut self, time: u64) -> anyhow::Result<()> {
+        let previous_slot = self.get_current_slot();
+
+        // Update store time
+        self.time = time;
+
+        let current_slot = self.get_current_slot();
+
+        // If this is a new slot, reset store.proposer_boost_root
+        if current_slot > previous_slot {
+            self.proposer_boost_root = B256::ZERO;
+        }
+
+        // If a new epoch, pull-up justification and finalization from previous epoch
+        if current_slot > previous_slot && compute_slots_since_epoch_start(current_slot) == 0 {
+            self.update_checkpoints(
+                self.unrealized_justified_checkpoint,
+                self.unrealized_finalized_checkpoint,
+            );
+        }
+
+        Ok(())
+    }
+
+    pub fn validate_target_epoch_against_current_time(
+        &mut self,
+        attestation: &Attestation,
+    ) -> anyhow::Result<()> {
+        let target = attestation.data.target;
+
+        // Attestations must be from the current or previous epoch
+        let current_epoch = self.get_current_store_epoch();
+        // Use GENESIS_EPOCH for previous when genesis to avoid underflow
+        let previous_epoch = if current_epoch > GENESIS_EPOCH {
+            current_epoch - 1
+        } else {
+            GENESIS_EPOCH
+        };
+        // If attestation target is from a future epoch, delay consideration until the epoch arrives
+        ensure!([current_epoch, previous_epoch].contains(&target.epoch));
+        Ok(())
```

Comment:
```suggestion

        Ok(())
```

It's a very little detail that makes code looks like well-cared.